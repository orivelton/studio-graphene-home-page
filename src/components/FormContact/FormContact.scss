@import '../../assets/scss/helper.scss';

.form {
  display: flex;;
  text-transform: uppercase;
  flex-direction: row;
  position: relative;
  &__box {
    width: 250px;
    display: flex;
    flex-direction: column;
    margin-right: 25px;
    &:last-of-type {
      margin-right: 0;
    }
  }

  &__label {
    color: $night-rider;
    font-size: 14px;
    margin: 10px 0 15px 0;
  }

  &__input {
    border: 0;
    border-bottom: 1px solid $night-rider;
    background: transparent;
    padding-bottom: 10px;
    &::placeholder {
      font-family: 'Gill Sans';
      color: $grey;
      font-size: 15px;
      opacity: 0.3;
      font-weight: 100;
    }
  }

  &__select {
    border: 0;
    border-bottom: 1px solid $night-rider;
    background: transparent;
    padding-bottom: 10px;
  }

  &__btn {
    border: 0;
    background: $night-rider;
    color: $cinderella;
    font-size: 18px;
    padding: 8.5px 16.6px 3.5px 16.6px;
    width: fit-content;
    font-weight: bold;
  }
}
