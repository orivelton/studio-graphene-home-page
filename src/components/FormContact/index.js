import React from 'react';
import './FormContact.scss';
import StyleSquare from '../StyleSquare';

const FormContact = () => (
  <form className="form">
    <span class="icon icondetail02_2x form__icon"></span>
    <div className="form__box">
      <label className="form__label" htmlFor="name">Name</label>
      <input className="form__input" id="name" type="text" placeholder="Name"/>

      <label className="form__label" htmlFor="phone">Phone</label>
      <input className="form__input" id="phone" type="tel" placeholder="phone"/>

      <label className="form__label" htmlFor="experience">Experience</label>
      <select id="experience" className="form__select">
        <option>Select</option>
      </select>
    </div>

    <div className="form__box">
      <label className="form__label" htmlFor="surname">Surname</label>
      <input className="form__input" id="surname" type="text" placeholder="Surname"/>

      <label className="form__label" htmlFor="email">EMAIL ADDRESS</label>
      <input className="form__input" id="email" type="email" placeholder="email"/>
      
      <label className="form__label" htmlFor="hear">Wherw did you hear about us?</label>
      <select id="hear" className="form__select">
        <option>Select</option>
      </select>
      <button className="form__btn uppercase">Send</button>
    </div>
  </form>
);

export default FormContact;
