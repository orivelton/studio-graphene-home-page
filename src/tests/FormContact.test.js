import React from 'react';
import renderer from 'react-test-renderer';
import FormContact from '../components/FormContact';

test('renders correctly FormContact', () => {
  const tree = renderer.create(<FormContact />).toJSON();
  expect(tree).toMatchSnapshot();
});
