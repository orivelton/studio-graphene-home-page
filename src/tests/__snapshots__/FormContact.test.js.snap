// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly FormContact 1`] = `
<form
  className="form"
>
  <span
    className="icon icondetail02_2x form__icon"
  />
  <div
    className="form__box"
  >
    <label
      className="form__label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="form__input"
      id="name"
      placeholder="Name"
      type="text"
    />
    <label
      className="form__label"
      htmlFor="phone"
    >
      Phone
    </label>
    <input
      className="form__input"
      id="phone"
      placeholder="phone"
      type="tel"
    />
    <label
      className="form__label"
      htmlFor="experience"
    >
      Experience
    </label>
    <select
      className="form__select"
      id="experience"
    >
      <option>
        Select
      </option>
    </select>
  </div>
  <div
    className="form__box"
  >
    <label
      className="form__label"
      htmlFor="surname"
    >
      Surname
    </label>
    <input
      className="form__input"
      id="surname"
      placeholder="Surname"
      type="text"
    />
    <label
      className="form__label"
      htmlFor="email"
    >
      EMAIL ADDRESS
    </label>
    <input
      className="form__input"
      id="email"
      placeholder="email"
      type="email"
    />
    <label
      className="form__label"
      htmlFor="hear"
    >
      Wherw did you hear about us?
    </label>
    <select
      className="form__select"
      id="hear"
    >
      <option>
        Select
      </option>
    </select>
    <button
      className="form__btn uppercase"
    >
      Send
    </button>
  </div>
</form>
`;
